File: problems/3507-minimum-pair-removal-to-sort-array-i/solution.py
class Solution:
    def minimumPairRemoval(self, nums: List[int]) -> int:
        operations = 0
        while True:
            min_sum = float('inf')
            min_index = -1

            for i in range(len(nums) - 1):
                current_sum = nums[i] + nums[i + 1]
                if current_sum < min_sum:
                    min_sum = current_sum
                    min_index = i

            if min_index == -1:
                is_sorted = True
                for i in range(len(nums) - 1):
                    if nums[i] > nums[i + 1]:
                        is_sorted = False
                        break
                if is_sorted:
                    return operations
                else:
                    return operations + len(nums) -1
            

            nums[min_index] = nums[min_index] + nums[min_index + 1]
            nums.pop(min_index + 1)
            operations += 1

File: problems/3507-minimum-pair-removal-to-sort-array-i/solution.c
int minimumPairRemoval(int* nums, int numsSize) {
    int operations = 0;
    while (1) {
        int minSum = 2001;
        int minIndex = -1;
        int sorted = 1;
        for (int i = 0; i < numsSize - 1; i++) {
            if (nums[i] > nums[i + 1]) {
                sorted = 0;
            }
            if (nums[i] + nums[i + 1] < minSum) {
                minSum = nums[i] + nums[i + 1];
                minIndex = i;
            }
        }

        if (sorted) {
            break;
        }

        if (minIndex == -1) break;

        nums[minIndex] = nums[minIndex] + nums[minIndex + 1];
        for (int i = minIndex + 1; i < numsSize - 1; i++) {
            nums[i] = nums[i + 1];
        }
        numsSize--;
        operations++;
    }
    return operations;
}